diff --git a/scripts/rl/calypsoEnv.py b/scripts/rl/calypsoEnv.py
index 84ef508..7c0f790 100644
--- a/scripts/rl/calypsoEnv.py
+++ b/scripts/rl/calypsoEnv.py
@@ -6,8 +6,8 @@ from tf_agents.environments import py_environment
 from scripts.const import *
 from scripts.tools.utils import*
 from scripts.pso.calypso import CALYPSO
-from scripts.functions import functions
 
+from copy import deepcopy
 
 
 def sigmoid(x):
@@ -194,8 +194,14 @@ class calypsoEnv(py_environment.PyEnvironment):
         self.funcGBN     = np.nan
         self.L_min      = self.funcRanges.loc[self.func, 'low']
         self.L_delta    = self.funcRanges.loc[self.func, 'delta']
-        funcAttr        = getattr(functions, self.func)
-        loss, lim, self.name = funcAttr()
+
+        funcClass = getattr(ela_benchmark_suite, self.func)
+        problem = funcClass(self.D)
+
+        loss = problem
+        lim = (problem.lower, problem.upper)
+        self.name = self.func
+
         self.funcName   = self.name
         self.range      = lim[1] - lim[0]
         self.calypso    = CALYPSO(
@@ -217,7 +223,7 @@ class calypsoEnv(py_environment.PyEnvironment):
 
         # Define runtime parameters
         self.i       = 1
-        self.lsbPrev = self.calypso.L_social_best.copy()
+        self.lsbPrev = deepcopy(np.array(self.calypso.L_social_best))
         self.lsbInit = self.lsbPrev.copy()
         vNormed      = np.zeros((self.N,), dtype=self.dtype)
         self.setState(self.i, 0.0, 0.5, vNormed)
